### ISSUE TITLE
Auto Accept folders always 'Up to date' after removed from remote device

### ISSUE THREAD

**Issue by memen45** [syncthing/syncthing#7917]
### Does your log mention database corruption?
No

### Include required information

 - which version of Syncthing and what operating system you are using
 v1.18.1, Windows (64-bit Intel/AMD)
"Fermium Flea"
Build 2021-07-30
 - what happened,
Enabled 'Auto Accept' feature. Added folders on device (Android). Folders were created automatically in Default folder. Now removed some folders from Syncthing on device (Android). Folders were not removed automatically and are now 'orphaned' directories. It is impossible to see which folders still have a connection with the device and which have not.
 - what you expected to happen instead,
Remove some folders from Syncthing on device (Android) and I would expect they remove from the list of folders as well. Or at least do not show 'Up to date' when in fact there is no remote to be up to date with?
 - any steps to reproduce the problem.
Add device and enable Auto Accept feature, set default folder. Add folders on remote device, wait for sync and remove folders. Now the removed ones remain in the overview and are marked as 'Up to date'.

**Comment by tomasz1986** [comment]
There may be a slight delay after removing the folders for the other side to be informed of the changes and change their state to "Unshared". Does the "Up to Date" state persist if you reset the connection (e.g. by pausing and unpausing one of the devices)?

**Comment by calmh** [comment]
"Auto accept" does not imply "auto remove" (which Syncthing doesn't have). If I have a folder, and the other side no longer shares it, I'm the only participant in that folder and I'm by definition "up to date".

**Comment by AudriusButkevicius** [comment]
Agreed, this is working as intended I think.

**Comment by memen45** [comment]
Thanks for your comments. @tomasz1986 Pausing/unpausing, rescanning, or even exiting and opening SyncTrayzor completely does not change the status. It remains Up to date. Also under device > folders, the folders are still shown.

**Comment by tomasz1986** [comment]
I have done the following.

1. Connect Device 1 and Device 2 with each other.
2. On Device 2, enable "Auto Accept" for Device 1.
3. On Device 1, create a new folder and share it with Device 2.
4. The folder has been automatically added to Device 2 at the default path.
5. On Device 1, remove the folder from Syncthing.
6. Reset the connection between the two devices.
7. Device 2 is trying to re-share the folder with Device 1.

![image](https://user-images.githubusercontent.com/5626656/131517916-97b337cc-d56f-4edf-aabc-0674afce0dcb.png)
![image](https://user-images.githubusercontent.com/5626656/131517926-a7023049-dbbd-45c9-b83f-7e35f71631f6.png)

Basically, the folder on Device 2 is marked as "Up to Date" and not "Unshared", because Device 2 still wants to share it with Device 1, regardless of the fact that Device 1 has removed/unshared the folder. If this is the case on your side too, then I would say that everything seems to be working as intended :relieved:. Is this correct?

**Comment by memen45** [comment]
You described my exact steps, thanks for your clarification! I understand now that it is intended behaviour. 

Seems like it is impossible then to manage shared folders on remote devices? Except you can of course stop sharing, but that only stops future changes.

**Comment by AudriusButkevicius** [comment]
No, management of remote folders is not a thing. The decentralised is the keyword in the description.

**Comment by memen45** [comment]
> No, management of remote folders is not a thing. The decentralised is the keyword in the description.

Well, this is violated by the Auto Accept feature then. It is explicitly there to let local folders be created by remote action only.

Now that the feature is more clear to me, maybe a feature request for 'Auto track' feature would be more logical. The folders from a remote device - now added through Auto Accept - can be grouped in the UI, so it is clear they are there as a result of the device Auto Track feature. Ideally (and for data reliability) if the device removed a folder, the folder can be marked as e.g. 'Unshared', maybe with the question 'Do you want to keep data here?'. 

Are you sure it requires protocol extension? If one can read the list of folders shared by a device, you can compare it with the local folders you have tracked for this device. Additions/removals can be detected based on this?
